File: dotNet/Tests/HebMorphTests/Lucene/HebrewQueryParserTest.cs

using Xunit;
using global::Lucene.Net.QueryParsers;
using global::Lucene.Net.QueryParsers.Hebrew;
using LuceneUtil = global::Lucene.Net.Util;

namespace HebMorph.Lucene.Tests
{
    public class HebrewQueryParserTest
    {
        [Fact]
        public void ParsesAcronymsCorrectly()
        {
            QueryParser qp = new HebrewQueryParser(LuceneUtil.Version.LUCENE_30, "f", new global::Lucene.Net.Analysis.Hebrew.SimpleAnalyzer());
            qp.Parse(@"צה""ל");
            qp.Parse(@"""צהל""");
            qp.Parse(@"כל הכבוד לצה""ל");
            qp.Parse(@"""כל הכבוד לצה""ל""");
            qp.Parse(@"""כל הכבוד"" לצה""ל");

            qp.Parse(@"מנכ""לית");

            try
            {
                qp.Parse(@"צה""""ל");
                qp.Parse(@"""צה""ל");
                Assert.True(false, "Expected exception was not thrown");
            }
            catch(ParseException) { }
        }
    }
}

File: java/hebmorph-lucene/src/test/java/org/apache/lucene/queryparsers/HebrewQueryParserTest.java
package org.apache.lucene.queryparsers;

import com.code972.hebmorph.hspell.HSpellLoader;
import org.apache.lucene.analysis.hebrew.SimpleAnalyzer;
import org.apache.lucene.queryparser.classic.ParseException;
import org.apache.lucene.queryparser.classic.QueryParser;
import org.apache.lucene.util.Version;
import org.junit.Test;

import java.io.IOException;

import static org.junit.Assert.fail;

public class HebrewQueryParserTest
{
    @Test
    public void ParsesAcronymsCorrectly() throws ParseException, IOException {

        QueryParser qp = new HebrewQueryParser(Version.LUCENE_46, "f", new SimpleAnalyzer(Version.LUCENE_46, HSpellLoader.readDefaultPrefixes()));
        qp.parse("צה\"ל");
        qp.parse("\"צהל\"");
        qp.parse("כל הכבוד לצה\"ל");
        qp.parse("\"כל הכבוד לצה\"ל\"");
        qp.parse("\"כל הכבוד\" לצה\"ל");

        qp.parse("מנכ\"לית");

        try
        {
            qp.parse("צה\"\"ל");
            qp.parse("\"צה\"ל");
            fail("Expected exception was not thrown");
        }
        catch(ParseException ex) { }
    }
}
